app/src/test/java/com/codesoom/assignment/application/ProductServiceTest.java

```diff
@@ -0,0 +1,155 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.repository.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@DisplayName("ProductService 클래스 테스트")
+class ProductServiceTest {
+
+    private CatToyService service;
+    private ProductRepository repository;
+
+    private static final String PRODUCT_NAME = "춘식이 고구마 장난감";
+    private static final String MAKER        = "카카오";
+    private static final Long PRICE          = 39000L;
+    private static final String IMAGE_URL    = "http://localhost:8080/snake";
+
+    private Product product = new Product(PRODUCT_NAME, MAKER, PRICE, IMAGE_URL);
+
+    @BeforeEach
+    void setUp() {
+        //TODO 왜 mock로 감싸야하는지 강의 다시보기!
+        repository = mock(ProductRepository.class);
+        service = new CatToyService(repository);
+
+        List<Product> products = new ArrayList<>();
+        products.add(product);
+
+        given(service.findAll()).willReturn(products);
+        given(service.findById(1L)).willReturn(Optional.of(product));
+        given(service.findById(100L)).willReturn(Optional.empty());
+    }
+
+    @Nested
+    @DisplayName("save 메소드는")
+    class Describe_save {
+        @Nested
+        @DisplayName("id가 존재하지 않으면")
+        class Context_without_id {
+            @Test
+            @DisplayName("product를 리포지토리에 저장한다")
+            void it_save_product() {
+                service.save(product);
+
+                verify(repository).save(product);
+
+                assertThat(service.findById(1L)).isNotNull();
+                assertThat(service.findById(1L)
+                                    .get()
+                                    .getProductName())
+                                    .isEqualTo(PRODUCT_NAME);
```

Comment:
```suggestion
                                    .isEqualTo(product.getName());
```

이렇게 만들어놓은 상품이랑 비교를 해서 `service.save(product)`와 같다는 것을 드러낼 수 있겠습니다